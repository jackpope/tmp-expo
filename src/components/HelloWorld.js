import { Text, View } from "react-native";

export default function HelloWorld() {
  return (
    <View testID="top">
      <Text>Hello</Text>
      <View testID="world-container">
        <Text>World</Text>
      </View>
    </View>
  );
}
