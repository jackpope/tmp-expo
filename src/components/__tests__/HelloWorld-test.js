import {ReactNativeTestRenderer} from '@react-native/test-renderer';
import * as React from 'react';
import * as ReactTestRenderer from 'react-test-renderer'
import HelloWorld from '../HelloWorld';
import { View, Text } from 'react-native';

describe('rtr', () => {
  it('returns expected JSON output based on renderer component', () => {
    const result = ReactTestRenderer.create(<HelloWorld />)
    expect(result.toJSON()).toMatchSnapshot()
  });
})

describe('render', () => {
  describe('toJSON', () => {
    it('returns expected JSON output based on renderer component', () => {
      const result = ReactNativeTestRenderer.render(<HelloWorld />);
      expect(result.toJSON()).toMatchSnapshot();
    });
  });

  describe('findByTestID', () => {
    it('returns the first element with a matching testID', () => {
      const result = ReactNativeTestRenderer.render(<HelloWorld />);
      expect(result.findByTestID('world-container')).not.toBeNull();
    });
  });

  describe('press', () => {
    it('invokes onPress callback', () => {
      let pressed = false;
      const result = ReactNativeTestRenderer.render(
        <View>
          <Text testID="text" onPress={() => (pressed = true)}>
            Hello
          </Text>
        </View>,
      );
      const text = result.findByTestID('text');
      expect(pressed).toBe(false);
      ReactNativeTestRenderer.fireEvent(text, 'press');
      expect(pressed).toBe(true);
    });
  });
});
