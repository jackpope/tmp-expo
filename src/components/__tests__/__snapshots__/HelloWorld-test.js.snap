// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render toJSON returns expected JSON output based on renderer component 1`] = `
"<>
  <RCTView>
    <RCTText accessible=true allowFontScaling=true ellipsizeMode="tail" isHighlighted=false selectionColor=null>
      <RCTRawText text="Hello" />
    </RCTText>
    <RCTView>
      <RCTText accessible=true allowFontScaling=true ellipsizeMode="tail" isHighlighted=false selectionColor=null>
        <RCTRawText text="World" />
      </RCTText>
    </RCTView>
  </RCTView>
</>"
`;

exports[`rtr returns expected JSON output based on renderer component 1`] = `
<RCTView
  testID="top"
>
  <RCTText
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    isHighlighted={false}
    selectionColor={null}
  >
    Hello
  </RCTText>
  <RCTView
    testID="world-container"
  >
    <RCTText
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      isHighlighted={false}
      selectionColor={null}
    >
      World
    </RCTText>
  </RCTView>
</RCTView>
`;
